### ISSUE TITLE
Update field when another field change

### ISSUE THREAD

**Issue by cdegrel** [pablo-abc/felte#96]
Hello,

This is my first time using your library that I find really interesting.

In my project, I use `solidjs` and `@felte/solid^1.0.0-next.18` to build a step form. I have a problem when I try to modify a field when another field changes.
I followed the documentation in the section https://next.felte.dev/docs/solid/migrating#stores, but in my console, I get the following error message `Uncaught InternalError: too much recursion`

I thank you in advance for your help. 

This is my code, if it can help solve my problem when i use `createEffect`

```tsx
const periodicities = [
  { label: 'Oneshot', value: 'none', name: 'Prestation ponctuelle' },
  { label: 'Mensuel', value: 'monthly', name: 'Abonnement mensuel' },
  { label: 'Trimestriel', value: 'trimestriel', name: 'Abonnement trimestriel', disabled: true },
  { label: 'Annuel', value: 'yearly', name: 'Abonnement annuel' },
]

const schema = z.object({
  periodicity: z.string().min(1),
  name: z.string().min(1)
})

type Values = z.infer<typeof schema>

type StepProps = {
  initialValues: Values
  onSubmit: (values: Values) => void
  onBack: (values: Values) => void
}

export const DeclinationPricingStep = ({ initialValues, onSubmit, onBack }: StepProps) => {
  const { form, data, errors, reset, setFields } = createForm<Values, ValidatorConfig>({
    onSubmit,
    initialValues
  })

  onMount(() => {
    reset()
  })

  createEffect(() => {
    setFields('name', periodicities.find(p => p.value === data('periodicity'))?.name)
  })

  return (
   <form ref={form} style={{ display: 'flex', 'flex-direction': 'column', gap: '1rem' }}>
    <FormGroup label="Engagement type" error={errors('periodicity')}>
      <Box gap="medium">
        <For each={periodicities}>
          {(periodicity) => (
            <RadioButton
              {...periodicity}
              name="periodicity"
              id={`periodicitiy:${periodicity.value}`}
              variant="custom"
            />
          )}
        </For>
      </Box>
      <FormField for="name" label="Nom" error={errors('name')}>
        <TextField name="name" />
      </FormField>
    </FormGroup>

     .......
   </form>
  )
}
```

**Comment by pablo-abc** [comment]
Thanks! Really good catch, I had not noticed this. I already found why this is happening (I forgot to use `untrack` somewhere), I'll be pushing a fix during the day. In the meantime you can use `on` to prevent the infinite loop if you want to keep trying v1:

```js
import { on, createEffect } from 'solid-js';

// In your component:
const periodicity = () => data('periodicity');
createEffect(on(periodicity, (per) => {
  setFields('name', periodicities.find(p => p.value === per?.name)
});
```

I'll let you know when I publish the fix! (And I'll close this then)

**Comment by cdegrel** [comment]
Thank you for your quick response 😄 , your solution works perfectly. 
It's a pleasure for me to test `felte`

**Comment by pablo-abc** [comment]
Hey, @cdegrel ! This should be fixed on the latest pre-release (1.0.0-next.21). I'm closing this since this is not happening anymore on my testing. But do let me know if anything else arises!
